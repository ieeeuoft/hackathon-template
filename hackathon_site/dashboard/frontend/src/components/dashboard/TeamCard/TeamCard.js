import React from "react";
import styles from "./TeamCard.module.scss";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import TitledPaper from "components/general/TitledPaper/TitledPaper";

const TeamCard = ({ members, teamCode, handleEditTeam }) => {
    const title = teamCode === "" ? "Team" : "Team " + teamCode;

    return (
        <TitledPaper title={title}>
            {members.map((member, i) => (
                <Container className={styles.name} key={i}>
                    <Typography variant="body2" noWrap>
                        {member}
                    </Typography>
                </Container>
            ))}

            <Container className={styles.lastRow}>
                <Button color="primary" onClick={handleEditTeam}>
                    Edit
                </Button>
            </Container>
        </TitledPaper>
    );
};

export default TeamCard;
