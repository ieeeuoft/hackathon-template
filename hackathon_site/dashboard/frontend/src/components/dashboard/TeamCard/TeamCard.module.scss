@import "./../../../assets/abstracts/mixins";
@import "./../../../assets/abstracts/variables";

.name {
    padding: 12px 20px;
}

.lastRow {
    @include flexPosition($jusCon: flex-end);
    padding-top: 6px;
}
