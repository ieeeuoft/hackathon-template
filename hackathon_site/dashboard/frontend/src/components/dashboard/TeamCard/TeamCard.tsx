import React from "react";
import styles from "./TeamCard.module.scss";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import TitledPaper from "components/general/TitledPaper/TitledPaper";
import Button from "@material-ui/core/Button";
import { useSelector } from "react-redux";
import { teamCodeSelector, teamMemberNamesSelector } from "slices/event/teamSlice";

interface TeamProps {
    handleEditTeam?: any;
}

export const TeamCard = ({ handleEditTeam }: TeamProps) => {
    const members = useSelector(teamMemberNamesSelector);
    const teamCode = useSelector(teamCodeSelector);

    const title = teamCode === null ? "No Team" : "Team " + teamCode;
    // waiting for the team store
    return (
        <TitledPaper title={title}>
            {members?.length
                ? members?.map((member, i) => (
                      <Container className={styles.name} key={i}>
                          <Typography variant="body2" noWrap>
                              {member}
                          </Typography>
                      </Container>
                  ))
                : null}
            <Container className={styles.lastRow}>
                <Button
                    color="primary"
                    onClick={handleEditTeam}
                    data-testid="teamCardBtn"
                >
                    Edit
                </Button>
            </Container>
        </TitledPaper>
    );
};

export default TeamCard;
