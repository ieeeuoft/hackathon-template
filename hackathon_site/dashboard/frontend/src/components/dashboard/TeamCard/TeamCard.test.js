import React from "react";
import TeamCard from "./TeamCard";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { fireEvent } from "@testing-library/react";

describe("TeamCard", () => {
    const members = ["Aki", "Graham", "Alex", "Raymond"];
    const teamCode = "PAS3NLQ3";

    test("render", () => {
        const { asFragment } = render(
            <TeamCard members={members} teamCode={teamCode} />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test("Calls handleEditTeam when the edit button is clicked", () => {
        const handleEditTeamSpy = jest.fn();
        const { getByText } = render(
            <TeamCard
                members={members}
                teamCode={teamCode}
                handleEditTeam={handleEditTeamSpy}
            />
        );
        const thebutton = getByText("Edit");
        fireEvent.click(thebutton);
        expect(handleEditTeamSpy).toHaveBeenCalled();
    });
});
