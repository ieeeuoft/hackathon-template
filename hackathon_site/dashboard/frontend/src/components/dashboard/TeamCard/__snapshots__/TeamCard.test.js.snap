// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamCard render 1`] = `
<DocumentFragment>
  <h2
    class="MuiTypography-root title MuiTypography-h2 MuiTypography-noWrap"
    data-testid="titledPaperTitle"
  >
    Team PAS3NLQ3
  </h2>
  <div
    class="MuiPaper-root paper MuiPaper-elevation2"
  >
    <div
      class="MuiContainer-root name MuiContainer-maxWidthLg"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap"
      >
        Aki
      </p>
    </div>
    <div
      class="MuiContainer-root name MuiContainer-maxWidthLg"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap"
      >
        Graham
      </p>
    </div>
    <div
      class="MuiContainer-root name MuiContainer-maxWidthLg"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap"
      >
        Alex
      </p>
    </div>
    <div
      class="MuiContainer-root name MuiContainer-maxWidthLg"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap"
      >
        Raymond
      </p>
    </div>
    <div
      class="MuiContainer-root lastRow MuiContainer-maxWidthLg"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Edit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
